core/src/main/kotlin/com/bumble/appyx/core/children/ChildNodeCreationManager.kt

```diff
@@ -105,8 +159,59 @@ internal class ChildNodeCreationManager<Routing : Any>(
         }
     }
 
+    private fun childBuildContext(savedState: SavedStateMap?): BuildContext =
+        BuildContext(
+            ancestryInfo = AncestryInfo.Child(parentNode),
+            savedStateMap = savedState,
+            customisations = customisations.getSubDirectoryOrSelf(parentNode::class),
+        )
+
+    private fun childEntry(
+        key: RoutingKey<Routing>,
+        savedState: SavedStateMap?,
+        suspended: Boolean,
+    ): ChildEntry<Routing> =
+        if (suspended) {
+            ChildEntry.Suspended(key, savedState)
+        } else {
+            ChildEntry.Initialized(
+                key = key,
+                node = parentNode
+                    .resolve(key.routing, childBuildContext(savedState))
+                    .build()
+            )
+        }
+
+    private fun ChildEntry<Routing>.initialize(): ChildEntry.Initialized<Routing> =
+        when (this) {
+            is ChildEntry.Initialized -> this
+            is ChildEntry.Suspended ->
+                ChildEntry.Initialized(
+                    key = key,
+                    node = parentNode.resolve(
+                        routing = key.routing,
+                        buildContext = childBuildContext(savedState),
+                    ).build()
+                )
+        }
+
+    @Suppress("ForbiddenComment")
+    private fun ChildEntry<Routing>.suspend(): ChildEntry.Suspended<Routing> =
+        when (this) {
+            is ChildEntry.Suspended -> this
+            is ChildEntry.Initialized ->
+                ChildEntry.Suspended(
+                    key = key,
+                    // TODO: Not able to get a scope from Compose here, providing fake one
```

Comment:
ticket? What's the impact?

Reply:
Depends on a client usage of `SaverScope`. We are providing `SaverScope` right now do be able to use `canSave` in code, but honestly I do not see any value in it right now because we are supporting Android only. Also not sure if anybody ever used `canSave` except Compose internals.